### ISSUE TITLE
Number of classes for Surrogate task

### ISSUE THREAD

**Issue by jlian2** [Confusezius/metric-learning-mining-interclass-characteristics#4]
I am still not sure how you cluster samples in surrogate task and how you specify number of labels ? I notice you are using "faiss", but could you please provide a bit more details?

**Comment by Confusezius** [comment]
All relevant clustering code is located in `auxiliaries.py` below lines 770. I embed all the training data, then subtract mean (optionally divide by standard deviation) and perform clustering (see `def deepcluster(...)`).

Hope that helps! I'll try to update the `README` to make that clearer once I have the time :).

**Comment by jlian2** [comment]
So is "number of clusters" a hyperparameter?

**Comment by Confusezius** [comment]
`shared_num_classes` is, which references the number of clusters (or shared/intraclass feature classes). See `Result_Runs.sh` for some example runs that set the key hyperparameters.
